src/daft-execution/src/executor/local/local_executor.rs

```diff
@@ -0,0 +1,167 @@
+use std::sync::{Arc, Mutex};
+
+use common_error::DaftResult;
+use daft_plan::ResourceRequest;
+use snafu::futures::TryFutureExt;
+use snafu::ResultExt;
+
+use crate::executor::{
+    resource_manager::{ExecutionResources, ResourceManager},
+    Executor,
+};
+use crate::partition::{virtual_partition::VirtualPartition, PartitionRef};
+use crate::task::Task;
+
+use super::local_partition_ref::LocalPartitionRef;
+
+#[derive(Debug)]
+pub struct LocalExecutor {
+    resource_manager: Arc<Mutex<ResourceManager>>,
+}
+
+impl LocalExecutor {
+    pub fn new(resource_capacity: ExecutionResources) -> Self {
+        let resource_manager = Mutex::new(ResourceManager::new(resource_capacity)).into();
+        Self { resource_manager }
+    }
+}
+
+impl Executor<LocalPartitionRef> for LocalExecutor {
+    fn can_admit(&self, resource_request: &ResourceRequest) -> bool {
```

Comment:
I get the potential issue of the scheduler misusing the current executor APIs, but just to be clear, the current scheduling protocol is:

Get all admissible tasks, i.e. tasks whose resource requests can fit into the currently available resources.
Use some scheduling policy to choose the best admissible task to submit next.
Submit task - this is where the resource request from the task is subtracted from the currently available resources.
I.e there's no concurrent submission on a set of admissible tasks. The fact that a given task is admissible should never be stale since:

We always submit a single task after gathering up all admissible tasks (no batch submission after the admissibility query).
The only concurrent access that could happen here is if a task finishes, in which case availability will be increased, which can't make admissible tasks inadmissible.
And we obviously don't want to hold the lock from can_admit to submit_task or prematurely decrement the available resources upon can_admit, since we want to get all admissible tasks and then choose the "best" one to submit next.